src/main/java/de/tum/in/www1/artemis/service/compass/CompassService.java

```diff
@@ -529,10 +529,12 @@ public void printStatistic(Long exerciseId) {
      */
     public List<Long> getCalculationEngineModelsWaitingForAssessment(Long exerciseId) {
         List<ModelingSubmission> modelingSubmissions = modelingSubmissionRepository.findSubmittedByExerciseIdWithEagerResultsAndFeedback(exerciseId);
-        assessAllAutomatically(modelingSubmissions.stream().map(Submission::getId).collect(Collectors.toList()), exerciseId);
+
         CompassCalculationEngine engine = compassCalculationEngines.get(exerciseId);
         engine.notifyNewModels(modelingSubmissions);
 
+        assessAllAutomatically(modelingSubmissions.stream().map(Submission::getId).collect(Collectors.toList()), exerciseId);
```

Comment:
why is the ordering of importance here?

Reply:
Engine needs to be notified about the new models before assessing them, there is a preparation step before assessment.